ts/WoltLabSuite/Core/Ui/Feed/Dialog.ts

```diff
@@ -0,0 +1,89 @@
+ *
+ * @author  Matthias Schmidt
+ * @copyright  2001-2021 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Feed/Dialog
+ */
+
+import UiDialog from "../Dialog";
+import * as StringUtil from "../../StringUtil";
+import * as Language from "../../Language";
+import User from "../../User";
+import * as DomTraverse from "../../Dom/Traverse";
+import * as Clipboard from "../../Clipboard";
+import * as UiNotification from "../Notification";
+
+/**
+ * Copies one of links to the clipboard.
+ */
+async function copy(event: Event): Promise<void> {
+  event.preventDefault();
+
+  const target = event.currentTarget as HTMLElement;
+  const input = DomTraverse.prevBySel(target, 'input[type="text"]') as HTMLInputElement;
+
+  await Clipboard.copyTextToClipboard(input.value);
+
+  UiNotification.show(Language.get("wcf.global.rss.copy.success"));
+}
+
+/**
+ * Opens the dialog with an anonymous and personalized link after clicking on the RSS feed link.
+ */
+function openDialog(event: Event): void {
+  event.preventDefault();
+
+  const alternative = event.currentTarget as HTMLAnchorElement;
+  const linkWithAccessToken = alternative.href;
+
+  const linkWithoutAccessToken = linkWithAccessToken
+    .replace(`at=${User.userId}-${User.accessToken}&`, "&")
+    .replace(new RegExp(`(\\?|&)at=${User.userId}-${User.accessToken}`), "");
+
+  UiDialog.openStatic(
+    "feedLinkDialog",
+    `
+<p class="info">${Language.get("wcf.global.rss.accessToken.info")}</p>
+<dl>
+  <dt>${Language.get("wcf.global.rss.withoutAccessToken")}</dt>
+  <dd>
+    <div class="inputAddon">
+      <input type="text" class="long" readonly value="${StringUtil.escapeHTML(linkWithoutAccessToken)}">
+      <a href="#" class="inputSuffix button jsTooltip feedLinkDialogCopyButton" title="${Language.get(
+        "wcf.global.rss.copy",
+      )}"><span class="icon icon16 fa-files-o pointer"></span></a>
+    </div>
+  </dd>
+</dl>
+<dl>
+  <dt>${Language.get("wcf.global.rss.withAccessToken")}</dt>
+  <dd>
+    <div class="inputAddon">
+      <input type="text" class="long" readonly value="${StringUtil.escapeHTML(linkWithAccessToken)}">
+      <a href="#" class="inputSuffix button jsTooltip feedLinkDialogCopyButton" title="${Language.get(
+        "wcf.global.rss.copy",
+      )}"><span class="icon icon16 fa-files-o pointer"></span></a>
+    </div>
+  </dd>
+</dl>
+`,
+    {
+      onSetup(content: HTMLElement) {
+        content
+          .querySelectorAll(".feedLinkDialogCopyButton")
+          .forEach((el) => el.addEventListener("click", (ev) => copy(ev)));
+      },
+      title: alternative.title || Language.get("wcf.global.button.rss"),
+    },
+  );
+}
+
+export function setup(): void {
+  document.querySelectorAll("a[rel=alternate]").forEach((link: HTMLAnchorElement) => {
+    if (new URL(link.href).searchParams.get("at") === `${User.userId}-${User.accessToken}`) {
```

Comment:
I'm not sure how I feel about specifically matching the user's accessToken vs. matching anything with `at` and is not an external URL.

Reply:
What be the advantage of matching anything else than the expected access token for the active user?